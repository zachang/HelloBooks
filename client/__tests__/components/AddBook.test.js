import React from 'react';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import storeConfiguration from '../../src/store/store';
import { AddBook } from '../../src/components/AddBook.jsx';

const event = {
  preventDefault: jest.fn(),
  target: {
    bookName: 'Charli Chapman',
    author: 'Creg Gadner',
    bookCount: 1,
    categoryId: 1,
    publishYear: new Date(),
    isbn: '#1000789',
    pages: 7000,
    description: 'mmmmmmm',
    bookImage: 'mmmm.jpg',
    bookImage_text: 'mmmmmm',
    bookContent: 'kkkkk.pdf',
    bookContent_text: 'mmmmm'
  }
};

describe('<AddBook/>', () => {
  // create a spy function for addBookAction
  const addBookAction = sinon.spy();
  // create a spy function for getCategoryAction
  const getCategoryAction = sinon.spy();
  // spy on handleSubmit of Signup Page
  sinon.spy(AddBook.prototype, 'handleSubmit');
  // spy on handleChange of Signup Page
  sinon.spy(AddBook.prototype, 'handleChange');
  // spy on componentWillMount of Signup Page
  sinon.spy(AddBook.prototype, 'componentWillMount');
  // spy on componentWillReceiveProps of Signup Page
  sinon.spy(AddBook.prototype, 'componentWillReceiveProps');

  const props = {
    bookState: {
      success: null,
      errors: null,
      passes: null
    },
    categoryState: {
      success: null,
      errors: null,
      fails: null
    },
    addBookAction,
    getCategoryAction
  };

  const middleware = [thunk];
  const mockStore = configureMockStore(middleware);
  let store;
  let shallowComponent;
  beforeEach(() => {
    store = mockStore(storeConfiguration);
    shallowComponent = shallow(<AddBook {...props} />);
  });

  it('renders <AddBook /> component', () => {
    const tree = toJson(shallowComponent);
    expect(tree).toMatchSnapshot();
  });
});