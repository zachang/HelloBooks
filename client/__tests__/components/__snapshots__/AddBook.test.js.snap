// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddBook/> renders <AddBook /> component 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col s10 m8 l6 bookadd"
      style={
        Object {
          "marginLeft": "35%",
          "marginTop": "4%",
        }
      }
    >
      <div
        className="row"
      >
        <form
          className="col s10"
          onSubmit={[Function]}
        >
          <div
            className="row"
          >
            <div
              className="input-field col s10"
            >
              <input
                className=""
                id="bookName"
                name="bookName"
                onChange={[Function]}
                type="text"
                value=""
              />
              <label
                className="custom-validate"
                data-error=""
                htmlFor="bookName"
              >
                Book Name
              </label>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="input-field col s10"
            >
              <input
                className=""
                id="author"
                name="author"
                onChange={[Function]}
                type="text"
                value=""
              />
              <label
                className="custom-validate"
                data-error=""
                htmlFor="author"
              >
                Author
              </label>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="input-field col s10"
            >
              <input
                className="datepicker"
                id="publishYear"
                name="publishYear"
                type="text"
                value=""
              />
              <label
                className="custom-validate"
                data-error=""
                htmlFor="publishYear"
              >
                Publish Year
              </label>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="input-field col s10"
            >
              <input
                className=""
                id="isbn"
                name="isbn"
                onChange={[Function]}
                type="text"
                value=""
              />
              <label
                className="custom-validate"
                data-error=""
                htmlFor="isbn"
              >
                ISBN
              </label>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="input-field col s10"
            >
              <input
                className=""
                id="pages"
                name="pages"
                onChange={[Function]}
                type="number"
                value=""
              />
              <label
                className="custom-validate"
                data-error=""
                htmlFor="pages"
              >
                Pages
              </label>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="input-field col s10"
            >
              <textarea
                className="materialize-textarea"
                id="description"
                name="description"
                onChange={[Function]}
              />
              <label
                className="custom-validate"
                data-error=""
                htmlFor="description"
              >
                Description
              </label>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="input-field col s10"
            >
              <select
                className=""
                id="categoryId"
                name="categoryId"
                value=""
              >
                <option
                  value=""
                >
                  Select
                </option>
              </select>
              <label
                className="custom-validate"
                data-error=""
                htmlFor="categoryId"
              >
                Choose Category
              </label>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="input-field col s10"
            >
              <input
                className=""
                id="bookCount"
                name="bookCount"
                onChange={[Function]}
                type="number"
                value=""
              />
              <label
                className="custom-validate"
                data-error=""
                htmlFor="bookCount"
              >
                Add Count
              </label>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="file-field input-field col s10"
            >
              <div
                className="btn"
              >
                <span>
                  Add Image
                </span>
                <input
                  accept="image/*"
                  id="bookImage"
                  name="bookImage"
                  onChange={[Function]}
                  type="file"
                />
              </div>
              <div
                className="file-path-wrapper"
              >
                <input
                  className="file-path"
                  name="bookImage_text"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="file-field input-field col s10"
            >
              <div
                className="btn"
              >
                <span>
                  Add PDF
                </span>
                <input
                  accept="application/pdf"
                  id="bookContent"
                  name="bookContent"
                  onChange={[Function]}
                  type="file"
                  value=""
                />
              </div>
              <div
                className="file-path-wrapper"
              >
                <input
                  className="file-path"
                  name="book_file_text"
                  type="text"
                />
              </div>
            </div>
          </div>
          <button
            className="col s10 btn btn-large waves-effect waves-light"
            name="action"
            type="submit"
          >
            Add Book
          </button>
        </form>
      </div>
    </div>
  </div>
  <div
    className="row imagePreview"
  />
</div>
`;
